### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by MosheWasserb**
Adding a new notebook demonstrates Zero cost Zero time Zero shot Financial Sentiment Analysis
From GPT4/Mixtral to MLP128K with SetFit 
@tomaarsen  Could you also send to Moritz Laurer for review?

### CODE REVIEW

**Comment by tomaarsen** [comment]
Looks promising at a glance! cc @MoritzLaurer

File: notebooks/Zero_cost_Zero_time_Zero_shot_Financial_Sentiment_Analysis.ipynb [hunk 1]
```diff
@@ -0,0 +1,2428 @@
+{
```
Comment (MoritzLaurer) [inline notebooks/Zero_cost_Zero_time_Zero_shot_Financial_Sentiment_Analysis.ipynb:1]: <p>would recommend versioning the key libraries to avoid issues with breaking changes in the future</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (MoritzLaurer) [inline notebooks/Zero_cost_Zero_time_Zero_shot_Financial_Sentiment_Analysis.ipynb:1]: <p>provide a bit more context where this figure comes from and what it represents. i.e. zeroshot for the 3 generative LLMs and a fine-tuned RoBERTa based on the zeroshot synthetic data from Mixtral (CoT + SC) (1800~ data rows/texts)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (MoritzLaurer) [inline notebooks/Zero_cost_Zero_time_Zero_shot_Financial_Sentiment_Analysis.ipynb:1]: <p>use consistent terminology: pseudo labels or synthetic data (or better: explain that pseudo labels are synthetic data)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (MoritzLaurer) [inline notebooks/Zero_cost_Zero_time_Zero_shot_Financial_Sentiment_Analysis.ipynb:1]: <p>do you mean "download" instead of "upload"?</p><p>would also slightly reformulate to make it clear that by "skip the training step" you mean skipping the code two cells further down. (could maybe even add an if else to let people choose)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (MoritzLaurer) [inline notebooks/Zero_cost_Zero_time_Zero_shot_Financial_Sentiment_Analysis.ipynb:1]: <p>installs should probably all be at the very beginning</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (MoritzLaurer) [inline notebooks/Zero_cost_Zero_time_Zero_shot_Financial_Sentiment_Analysis.ipynb:1]: <p>interesting, didn't know the WOS approach/metric. I suppose that word order is only one thing that transformers take into account. Another aspect would be semantic (dis)similarity of different strings. With a countvectorizer you only capture the exact words that are in the training corpus, but it can't capture semantically similar words that are outside the training data distribution</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (MoritzLaurer) [inline notebooks/Zero_cost_Zero_time_Zero_shot_Financial_Sentiment_Analysis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #11.    print('The WOS implies that in average {:0.1f}% of the sentences in the financial sentiment analysis (FSA) dataset are rather simple.\n'.format(100-100*WOS))
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (MoritzLaurer) [inline notebooks/Zero_cost_Zero_time_Zero_shot_Financial_Sentiment_Analysis.ipynb:1]: <p>typo "prbabilities". </p><p>I would maybe also add a note somewhere that this is less likely to work on more complex reasoning tasks (I'd assume e.g. that countvectorizers just can't represent complex semantics / classes well enough)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (MoritzLaurer) [inline notebooks/Zero_cost_Zero_time_Zero_shot_Financial_Sentiment_Analysis.ipynb:1]: <p>interesting! (worth noting that you are also increasing the size of the MLP here, in addition to adding more training data. maybe make that (small) increase in size explicit)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/497/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

**Comment by MoritzLaurer** [comment]
Looks interesting and good to me. Would assume that this works less well for more complex tasks and the additional step of distilling from the setfit model takes more developer time, but overall a cool approach for further compressing the model and making things much more efficient for inference

**Comment by MoritzLaurer** [comment]
> Thanks @MoritzLaurer for the comments. We updated the notebook accordingly. Would you be interested in a joint post/blog or expanding your original blog with the MLP example?

Great! Don't have bandwidth for a joint blog atm unfortunately. 
Notebook LGTM @tomaarsen

**Comment by MosheWasserb** [comment]
Hi @tomaarsen  Could you merge into main?

**Comment by tomaarsen** [comment]
@MosheWasserb My apologies for the radio silence here, I was very busy with https://github.com/UKPLab/sentence-transformers/releases/tag/v2.6.0
Very impressive performance on this work.

**Comment by MosheWasserb** [comment]
Hi @tomaarsen Sure, no problem.
Great work with the binary embeddings.
Did you know that for SetFit I was able to compress a 768-vector size into 2 dim with no accuracy loss?

**Comment by tomaarsen** [comment]
Thank you! PCA remains strong indeed, especially for classification. It doesn't work very well for retrieval however, there I've had more luck with 1. Matryoshka models and 2. Quantization to speed up the comparisons.
